File: ios/event/MotionChangeEvent.m
//
//  MotionChangeEvent.m
//  RNBackgroundLocation
//
//  Motion Change Event
//  Android MotionChangeEvent.java benzeri
//

#import "MotionChangeEvent.h"
#import "LocationModel.h"

@implementation MotionChangeEvent

- (instancetype)initWithIsMoving:(BOOL)isMoving location:(LocationModel *)location {
    self = [super init];
    if (self) {
        _isMoving = isMoving;
        _location = location;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [[self.location toDictionary] mutableCopy];
    dict[@"isMoving"] = @(self.isMoving);
    return dict;
}

- (NSString *)getEventName {
    return @"motionchange";
}

@end

File: android/src/main/java/com/backgroundlocation/event/MotionChangeEvent.java
package com.backgroundlocation.event;

import org.json.JSONObject;

/**
 * Motion Change Event
 * Hareket durumu değiştiğinde tetiklenir
 */
public class MotionChangeEvent {
    private final boolean isMoving;
    private final JSONObject location;
    
    public MotionChangeEvent(boolean isMoving, JSONObject location) {
        this.isMoving = isMoving;
        this.location = location;
    }
    
    public boolean isMoving() {
        return isMoving;
    }
    
    public JSONObject getLocation() {
        return location;
    }
    
    public JSONObject toJson() {
        try {
            JSONObject json = new JSONObject();
            json.put("isMoving", isMoving);
            json.put("location", location);
            return json;
        } catch (Exception e) {
            return new JSONObject();
        }
    }
    
    public String getEventName() {
        return "motionchange";
    }
}



